### ISSUE TITLE
Wrong docstrings for tiles

### ISSUE THREAD

**Issue by jankatins** [SciTools/cartopy#750]
I'm trying out the different tiles in img_tiles and while using `QuadtreeTiles` I found that the docstring/signature does not fit the output:

`tiles = QuadtreeTiles(style="terrain")` gives no different output that the default "street". I suspect that this is simple because the docstring is from the "GoogleTiles" and as the constructor is not overwritten, the docstring stays the same.

Docstring/signature for the quadtree tiles as shown in the jupyter notebook:

```
Init signature: QuadtreeTiles(self, desired_tile_form='RGB', style='street')
Docstring:
Implements web tile retrieval using the Microsoft WTS quadkey coordinate
system.

A "tile" in this class refers to a quadkey such as "1", "14" or "141"
where the length of the quatree is the zoom level in Google Tile terms.
Init docstring:
:param desired_tile_form:
:param style: The style for the Google Maps tiles. One of 'street',
    'satellite', 'terrain', and 'only_streets'.
    Defaults to 'street'.
```

[Jupyter notebook, py3.5, cartopy 0.14 from conda-forge channel]

**Comment by pelson** [comment]
Right. The docstrings are off because the inheritance is a little bit squiffy in the img_tiles module (https://github.com/SciTools/cartopy/blob/master/lib/cartopy/io/img_tiles.py#L292). I think we probably need to add another superclass which the GoogleTiles and QuadtreeTiles each subclass.

**Comment by QuLogic** [comment]
The documentation inconsistency appears to have been corrected by the refactoring in #1188.
